be/src/exec/pipeline/scan/chunk_buffer_limiter.h

```diff
@@ -0,0 +1,71 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <mutex>
+
+namespace starrocks::pipeline {
+
+class ChunkBufferLimiter;
+using ChunkBufferLimiterPtr = std::unique_ptr<ChunkBufferLimiter>;
+
+class ChunkBufferLimiter {
+public:
+    virtual void update_avg_row_bytes(size_t added_sum_row_bytes, size_t added_num_rows) = 0;
+
+    virtual bool acquire_chunk(int num_chunks) = 0;
```

Comment:
Maybe `pin` and `unpin` is better, it's a name from MySQL's buffer-pool manager.